README.md

```diff
@@ -1,87 +1,105 @@
 # buildkite_agent
 
-Welcome to your new module. A short overview of the generated parts can be found in the PDK documentation at https://puppet.com/pdk/latest/pdk_generating_modules.html .
+![](https://github.com/call/puppet-buildkite_agent/workflows/MacOS%20QA/badge.svg)
 
-The README template below provides a starting point with details about what information to include in your README.
+A Puppet module to manage [Buildkite Agent](https://buildkite.com/docs/agent/v3) on macOS.
 
-#### Table of Contents
+## Table of Contents
 
-1. [Description](#description)
-2. [Setup - The basics of getting started with buildkite_agent](#setup)
-    * [What buildkite_agent affects](#what-buildkite_agent-affects)
-    * [Setup requirements](#setup-requirements)
-    * [Beginning with buildkite_agent](#beginning-with-buildkite_agent)
-3. [Usage - Configuration options and additional functionality](#usage)
-4. [Limitations - OS compatibility, etc.](#limitations)
-5. [Development - Guide for contributing to the module](#development)
+- [buildkite_agent](#buildkiteagent)
+  - [Table of Contents](#table-of-contents)
+  - [Description](#description)
+  - [Setup](#setup)
+    - [Beginning with buildkite_agent](#beginning-with-buildkiteagent)
```

Comment:
```suggestion
    - [Beginning with buildkite_agent](#beginning-with-buildkite_agent)
```

Reply:
Good eyes! 👀 As it turns out though, underscores characters are [not recognized in markdown heading anchors](https://gist.github.com/asabaylus/3071099#gistcomment-1593627). News to me! 🗞

Reply:
Dang, it works in the master (that's why I placed them in the suggestion).

Reply:
Of note, though `pdk new module` generates markdown anchors in the TOC with underscores, VSCode automatically removes the underscores upon save. ✨

Reply:
Oh hahaha, that makes sense.

Reply:
@philipdelorenzo prodded me further on this, and it turns out that there was some overly-aggressive VSCode auto-formatting happening.

Underscores in heading anchors == 👍